feature_engine/selection/drop_high_PSI_features.py

```diff
@@ -0,0 +1,460 @@
+        numerical variables in the dataset.
+
+    switch_basis: boolean, default=False.
+        If set to true the role of the two matrices involved in the PSI calculations
+        (basis and measurement) will be switch. This is an important option as the
+        PSI value is not symmetric (i.e. PSI(a, b) != PSI(b, a)).
+
+    threshold: float, default = 0.25.
+        Threshold above which the distribution of a feature has changed so much that
+        the feature will be dropped. The most common values are 0.25 (large shift)
+        and 0.10 (medium shift).
+
+    method: string or callable, default='equal_frequency'
+        Type of binning used to represent the distribution of the feature. In can be
+        either "equal_width" for equally spaced bins or "equal_frequency" for bins
+        based on quantiles.
+
+    n_nbins, int, default = 10
+        Number of bins used in the binning. For numerical feature a value of 10 is
+        considered as appropriate. For features with lower cardinality lower values
+        are usually used.
+
+    min_pct_empty_buckets, float, default = 0.0001
+        Value to add to empty bucket (when considering percentages). If a bin is
+        empty the PSI value may jump to infinity. By adding a small number to empty
+        bins, this issue is avoided. If the value added is too large, it may disturb
+        the calculations.
+
+    missing_values: str, default=ignore
+        Takes values 'raise' and 'ignore'. Whether the missing values should be raised
+        as error or ignored when determining correlation.
+
+    Attributes
+    ----------
+    features_to_drop_:
+        Set with the correlated features that will be dropped.
+
+    correlated_feature_sets_:
+        Groups of correlated features. Each list is a group of correlated features.
+
+    variables_:
+        The variables to consider for the feature selection.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    psi:
+        Dataframe containing the PSI values for all features considered.
+
+    features_to_drop_:
+        List of features that are dropped because of PSI values above threshold.
+
+    Methods
+    -------
+    fit:
+        Find correlated features.
+    transform:
+        Remove correlated features.
+    fit_transform:
+        Fit to the data. Then transform it.
+    _compute_psi:
+        Perform binning and compute the PSI for all features
+    _compute_feature_psi:
+        Combine information and call psi calculation
+    _psi
+        Compute the PSI value
+    _split_dataframe:
+        returns the basis and the measurement dataframes needed for PSI calculations
+    _check_init_values:
+        Perform a series of check on the attributes.
+    """
+
+    # TODO: Implement the check on the types of the cut-off dates and the date
+    # column that need ot be the same.
+    def __init__(
+        self,
+        basis,
```

Comment:
Thanks. Maybe this is an option.
- Change the name basis into split, the default value is 0.5
- Have a column to sort by, default would be the index.
- If there are repeated values in the column used for sorting, computed the 'most natural split' to ensure that all observations with the same value in the sorted column belong to the same split.

For example:

split = 0.5 on the date columns for this dataframe

| Index   | Date   | Feature value |
|:-------|:------|:--------|
| 0         | 1 Jan 2020 | 27 |
| 1         | 2 Jan 2020 | 13 |
| 2         | 3Jan 2020 | 97 |
| 3         | 3 Jan 2020 | 71 |
| 4         | 4 Jan 2020 | 54 |
| 5         | 5 Jan 2020 | 32 || 

would lead to 

| Index   | Date   | Feature value |
|:-------|:------|:--------|
| 0         | 1 Jan 2020 | 27 |
| 1         | 2 Jan 2020 | 13 |
| 2         | 3Jan 2020 | 97 |
| 3         | 3 Jan 2020 | 71 |

and 

| 4         | 4 Jan 2020 | 54 |
| 5         | 5 Jan 2020 | 32 || 

Having the two values for 3 Jan in the same 'split'